### PR TITLE
cephadm: set pids-limit unlimited for all ceph daemons

### PR DESCRIPTION
**Pull request by adk3798**
We actually had this setup before, but ran into issues. Some teuthology test had failed in the fs suite, so it was modified to only affect iscsi and rgw daemons (https://github.com/ceph/ceph/pull/45798) and then the changes were reverted entirely (so no pids-limit modifying code at all) in quincy and pacific because the LRC ran into issues with the change related to the podman version (https://github.com/ceph/ceph/pull/45932). This new patch now addresses the podman versions, specifically that the patch that makes -1 work for a pids-limit seems to have landed in podman 3.4.1 based on https://github.com/containers/podman/pull/12040. We'll need to make sure that this doesn't break anything in the fs suites again as I don't remember the details of the first issue, or why having it only set the pids-limit for iscsi and rgw fixes it. Assuming that isn't a problem we should hopefully be able to unify at least how reef and quincy handle this now that the podman version issue is being addressed in this patch.

See the linked tracker issue for a discussion on why we're going at this again and why I'm trying to do this for all ceph daemon types.

Fixes: https://tracker.ceph.com/issues/58685

Signed-off-by: Adam King <adking@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/cephadm/cephadm.py [hunk 1]
```diff
@@ -3138,13 +3143,14 @@ def get_container(ctx: CephadmContext,
         envs.append('TCMALLOC_MAX_TOTAL_THREAD_CACHE_BYTES=134217728')
     if container_args is None:
         container_args = []
+    if daemon_type in Ceph.daemons or daemon_type in Ceph.gateways:
```
Comment (idryomov) [inline src/cephadm/cephadm.py:3146]: With all daemons and both iscsi and nfs services included, what entities managed by cephadm is this check meant to exclude?

Comment (adk3798) [reply src/cephadm/cephadm.py:3146]: anything not using a ceph container. Grafana, alertmanager, node-exporter, prometheus, keepalived, haproxy, SNMP, loki, promtail, and we also support custom containers that would be excluded here. `Ceph.daemons` is only
```
('mon', 'mgr', 'osd', 'mds', 'rgw', 'rbd-mirror', 'crash', 'cephfs-mirror', 'ceph-exporter')
```

File: src/cephadm/cephadm.py [hunk 2]
```diff
@@ -3187,7 +3193,6 @@ def get_container(ctx: CephadmContext,
         # So the container can modprobe iscsi_target_mod and have write perms
         # to configfs we need to make this a privileged container.
         privileged = True
-        set_pids_limit_unlimited(ctx, container_args)
```
Comment (idryomov) [inline src/cephadm/cephadm.py:3190]: The comment above in this branch ("Applies only on rbd-target-api as get_tcmu_runner_container() removes all tcmu-runner arguments") is obsolete, right?  It looks like it should have been removed in https://github.com/ceph/ceph/pull/46506.

Comment (adk3798) [reply src/cephadm/cephadm.py:3190]: yeah, seems like something that was added when setting the pids limit was being done differently earlier in the review process on the PR you linked. Now removing it as part of the commit here.

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
>       * [x]  No tests

There is an existing test for both iSCSI containers, let's make sure it still passes.

**Review by vumrao — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-02-21_05:38:18-orch:cephadm-wip-adk-testing-2023-02-20-1650-distro-default-smithi/

failed/dead job reruns: https://pulpito.ceph.com/adking-2023-02-24_17:44:54-orch:cephadm-wip-adk-testing-2023-02-20-1650-distro-default-smithi/

After reruns, 3 failures and 1 dead job

- dead job was failure pulling a podman package
```
Failed to download packages: podman-docker-3:4.3.1-2.module_el8.8.0+1254+78119b6e.noarch:
  Cannot download, all mirrors were already tried without success
 ```
   Another instance of this test passed in original run so didn't bother with another rerun

- 2 failures were staggered upgrade test issue tracked by https://tracker.ceph.com/issues/58535
- last failure was in currently flaky test_nfs task test. Passed when I tried an interactive rerun for debugging

Overall, nothing to block merging. Will note initial version of basic monitoring stack test passed.



> > ```
> >   * [x]  No tests
> > ```
> 
> There is an existing test for both iSCSI containers, let's make sure it still passes.

the iscsi pids limit test passed in the reruns (and initial failure was teuthology/infra issue) https://pulpito.ceph.com/adking-2023-02-24_17:44:54-orch:cephadm-wip-adk-testing-2023-02-20-1650-distro-default-smithi/7186371

**Pull request merged**